pkg/apis/admissionregistration/types.go

```diff
@@ -123,6 +123,27 @@ type ValidatingAdmissionPolicy struct {
 	metav1.ObjectMeta
 	// Specification of the desired behavior of the ValidatingAdmissionPolicy.
 	Spec ValidatingAdmissionPolicySpec
+	// The status of the ValidatingAdmissionPolicy, including warnings that are useful to determine if the policy
+	// behaves in the expected way.
+	// +optional
+	Status ValidatingAdmissionPolicyStatus `protobuf:"bytes,3,opt,name=status"`
+}
+
+// ValidatingAdmissionPolicyStatus represents the status of a ValidatingAdmissionPolicy.
+type ValidatingAdmissionPolicyStatus struct {
+	// ExpressionWarnings contains helpful information for expressions that can aid troubleshooting.
+	// +optional
+	// +listType=atomic
+	ExpressionWarnings []ExpressionWarning `protobuf:"bytes,1,rep,name=expressionWarnings"`
+}
+
+// ExpressionWarning is a warning information that targets a specific expression.
+type ExpressionWarning struct {
+	// Expression is the CEL expression that this warning is associated.
+	Expression string `protobuf:"bytes,1,opt,name=expression"`
```

Comment:
There will be multiple fields containing expressions in this release. If we get them all in we will have:

- `spec.validations[].expression`
- `spec.validations[].messageExpression`
- `spec.auditAnnotations[].valueExpression`
- `spec.matchConditions[].expression`

 I was going to suggest we add another field to `ExpressionWarning` to clarify which one of these fields the warning is about. This might be needed in the future because some expressions might have access to more variables than others (so an identical expression could be valid in one place but not another).

However, we don't have an immediate need. So maybe we just do nothing now and if we need to clarify we can add an enum field in a future release (e.g. `expressionType: AuditAnnotation`).

WDYT @cici37 ?

Reply:
This is a great point! Even the expressions are expected to be different in those fields, we will return one warning if they happen to have same expression. I am ok with having warning returning per expression as of now since we want to have a better understanding on the enum fields before adding them.

Reply:
The `refField` field is added to the `ExpressionWarning` type to indicate which one of these refers to the expression in question.